config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,114 @@
+class Rack::Attack
+
+  ### Configure Cache ###
+
+  # If you don't want to use Rails.cache (Rack::Attack's default), then
+  # configure it here.
+  #
+  # Note: The store is only used for throttling (not blacklisting and
+  # whitelisting). It must implement .increment and .write like
+  # ActiveSupport::Cache::Store
+
+  # Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new
+
+  ### Throttle Spammy Clients ###
+
+  # If any single client IP is making tons of requests, then they're
+  # probably malicious or a poorly-configured scraper. Either way, they
+  # don't deserve to hog all of the app server's CPU. Cut them off!
+
+  # Throttle all requests by IP
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.ip}"
+  throttle('req/ip',
+           limit: (ENV['REQUESTS_PER_FIVE_MINUTES'] || 500).to_i,
+           period: 5.minutes) do |req|
```

Comment:
Five minutes is kind of harsh. Also, not critical, but I feel like maybe the period could be an env var, too? That way we can adjust without having to do a redeploy.

Reply:
Hmm. I'm not sure I understand. The 5 minutes gives the ability to have a burst of activity during a 5 minute window. For example, 100 requests per minute would get throttled at 101 requests in one minute. But 500 requests in 5 minutes, the same total requests allowed over 5 minutes, would not throttle that 101st request. It would allow until the 501st request over 5 minutes.

Reply:
However, I do agree that period should be an ENV to allow for this to be adjustable in production without code changes. 👍

Reply:
I guess my reasoning is that the point of throttling anonymous access is to make sure we still retain some control over server resources. A five minute cooldown seems punitive.

Reply:
Interesting distinction. I see your point, but I was thinking it was less punitive to apply the throttle less frequently by allowing for bursts. Either way, I think our intent is to try to set the level high enough to rarely throttle anyone that isn't being either kinda dumb or intentionally abusive. I'll be making the period ENV configurable so we can setup reports in our logging to look for this happening in production and adjust until we feel we've got the right numbers for both values. 👍

Reply:
Yeah, good idea to check the logs to see how often throttling happens.